### ISSUE TITLE
Watching variable values in Debug

### ISSUE THREAD

**Issue by Gregory1701** [codefori/vscode-ibmi#1926]
I finally got debug in code4i to work! Unfortunately, the very first program I wanted to debug, which is a very small program just to see the different behaviors of RPGLE. Here is my source:

```
**Free
Ctl-Opt Main(BIF_Proc_Demo);
Ctl-Opt Debug Option(*SrcStmt:*NoDebugIO);
Ctl-Opt ActGrp(*Caller);

dcl-ds PgmDs psds qualified;
  ProcName *Proc;
end-ds;

dcl-s ProcedureName char(256);

dcl-proc BIF_Proc_Demo;

  dcl-s error_occurred ind Inz(*On);

  ProcedureName = %Proc();

  if error_occurred;
    dump(a);
  endif;

On-Exit error_occurred;

  if error_occurred;
    //dump(a);
  endif;

end-proc BIF_Proc_Demo;
```

And here is what I'm seeing in debug: 
![image](https://github.com/codefori/vscode-ibmi/assets/132368707/809f2cbf-f9aa-4099-8d2f-29549f6e275b)

Why doesn't my variable ProcedureName show me it's current value in the Watch section? It shows me the value down in the Debug Console.
<hr />

|Context|Version|
|-|-|
|Code for IBM i version|2.8.0|
|Visual Studio Code version|1.87.1|
|Operating System|linux_x64|

<details>
<summary>Active extensions</summary>

```
Code for IBM i Walkthroughs (vscode-ibmi-walkthroughs): 0.4.0
Db2 for IBM i (vscode-db2i): 0.8.0
Emmet (emmet): 1.0.0
Git (git): 1.0.0
Git Base (git-base): 1.0.0
GitHub (github): 0.0.1
GitHub Copilot (copilot): 1.172.0
GitHub Copilot Chat (copilot-chat): 0.13.0
IBM i Debug (ibmidebug): 1.0.0
JSON Language Features (json-language-features): 1.0.0
Merge Conflict (merge-conflict): 1.0.0
Node Debug Auto-attach (debug-auto-launch): 1.0.0
Prettier - Code formatter (prettier-vscode): 10.1.0
RPGLE (vscode-rpgle): 0.26.2
Server Ready Action (debug-server-ready): 1.0.0
TypeScript and JavaScript Language Features (typescript-language-features): 1.0.0
```
</details>

<hr />

<details>
<summary>Remote system</summary>

|Setting|Value|
|-|-|
|IBM i OS|V7R3M0|
|Tech Refresh|n/a|
|CCSID|37|
|Default CCSID|37|
|SQL|Enabled
|Source dates|Disabled

### Enabled features

|/QOpenSys/pkgs/bin|/usr/bin|/QSYS.lib/ILEDITOR.lib|/QSYS.LIB|/QIBM/ProdData/IBMiDebugService/bin|
|-|-|-|-|-|
|bash|attr|GENCMDXML.PGM|QZDFMDB2.PGM|startDebugService.sh|
|git|iconv|GETNEWLIBL.PGM|||
|grep|setccsid||||
|ls|||||
|md5sum|||||
|sort|||||
|stat|||||
|tar|||||
|tn5250|||||
</details>

<details>
<summary>Shell env</summary>

```bash
```
</details>

<details>
<summary>Variants</summary>

```json
{
  "american": "#@$",
  "local": "#@$"
}
```
</details>

<details>
<summary>Errors</summary>

```json
[
  {
    "command": "/QOpenSys/usr/bin/qsh",
    "code": 1,
    "stderr": "CPFA0A9:  Object not found.  Object is /QSYS.LIB/PGMGREGORY.LIB/VSCODE.FILE/RPGLINT.MBR.\nCPFA097:  Object not copied.  Object is /QSYS.LIB/PGMGREGORY.LIB/VSCODE.FILE/RPGLINT.MBR.",
    "cwd": "/home/PGMGREGORY"
  },
  {
    "command": "/QOpenSys/usr/bin/qsh",
    "code": 1,
    "stderr": "CPFA0A9:  Object not found.  Object is /QSYS.LIB/PGMGREGORY.LIB/VSCODE.FILE/RPGLINT.MBR.\nCPFA097:  Object not copied.  Object is /QSYS.LIB/PGMGREGORY.LIB/VSCODE.FILE/RPGLINT.MBR.",
    "cwd": "/home/PGMGREGORY"
  },
  {
    "command": "export BUILDLIB=\"PGMGREGORY\" && export CURLIB=\"PGMGREGORY\" && export USERNAME=\"pgmgregory\" && export HOST=\"prismpcr.pcrichard.com\" && export HOME=\"/home/PGMGREGORY\" && export WORKDIR=\"/home/PGMGREGORY\" && export TGTRLS=\"V7R2M0\" && export LIBLS=\"QGPL DISKNYN DISK RICHARD DTAGREGORY PGMGREGORY QTEMP\" && env",
    "code": 1,
    "stderr": "bsh: BUILDLIB=PGMGREGORY: is not an identifier",
    "cwd": "/home/PGMGREGORY"
  }
]
```
</details>

**Comment by sebjulliand** [comment]
I'll let @worksofliam confirm but I'm quite sure this has to do with how the debugger is managed by the debug extension, provided by IBM.

**Comment by worksofliam** [comment]
@sebjulliand You are absolutely correct.

I will pass this onto the debug team.

**Comment by mkwan01** [comment]
I can see the value of the variable ProcedureName in the Watch view:
![image](https://github.com/codefori/vscode-ibmi/assets/44405167/107a5347-936d-4291-95c6-3a5d7600e1f0)

This may have something to do with the compile options. You can debug the same program in STRDBG and enter "EVAL PROCEDURENAME" when you are at line 18. If you have RDi, you can also debug the same program in RDi and see whether you can monitor the same variable correctly. Alternatively, you can also send us the compile options for the program.

**Comment by mkwan01** [comment]
We just released a new debug client (v2.0.2) and a new debug service PTF. We can close this one now. Please reopen if you are seeing similar issues in the new debug client.

@worksofliam
